src/server/utils/settings-manager/settings-manager.ts

```diff
@@ -230,9 +237,9 @@ export class SettingsManager {
 
     logger.log(`Got external dataset removal for ${dataCubeName} in cluster ${cluster.name}`);
 
-    let dataCube = this.appSettings.getDataCube(dataCubeName);
+    let dataCube = getDataCube(this.appSettings, dataCubeName);
     if (dataCube) {
-      this.appSettings = this.appSettings.deleteDataCube(dataCube);
+      this.appSettings = deleteDataCube(this.appSettings, dataCube);
```

Comment:
Should we delete DataCubes or mark it as deleted?

Reply:
Delete. Marking as deleted definitely to complex for "refactoring" PR.